### DISCUSSION TITLE
Dealing with 1MB size limitation of the data URL when chatting with images (or other files)

### DISCUSSION THREAD

**Discussion by gmantri** [microsoft/semantic-kernel#6897] [General]
When passing an image to Azure OpenAI, we can pass it in 2 ways:

1. Give a publicly accessible URL for the image, or
2. Use the data URL of the image

The problem with 1st one is that at times it is just not possible to share the URL of an image file (for example, our corporate policy prohibits sharing OneDrive file outside of our organization) and the problem with 2nd one is that the data URL has a size limitation and cannot exceed 1MB.

I am curious to know how folks are dealing with these limitations. Any insights or workarounds are highly appreciated.

**Comment by dluc** [comment]
Hi @gmantri. Here's some options I would consider if I started to hit the 1MB threshold:

1. Uploading images to Azure Blobs, and obtaining a download URI that includes a token with expiration, e.g. 10 minutes. Pass that URL to OpenAI.
2. If Azure OpenAI and Azure Blobs are in the same region, it might be possible to use internal URI, better for security, something worth checking.
3. Compress the image and reduce the pixel size. If I remember correctly OneDrive provides multiple views of the same image, e.g. one can download the same image at different resolutions. 
4. Use some opensource tool or library to compress imaged on the fly.

**Reply by gmantri** [reply]
@dluc - Many thanks for the response. I am wondering if using Private Link between Azure OpenAI and Azure Storage Account is an option.
